src/lib/components/DashNgl.react.js

```diff
@@ -0,0 +1,219 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Stage, Selection} from 'ngl';
+
+/**
+ * The NGL component is used to render schematic diagrams
+ * of biomolecules in ribbon-structure representations.
+ * Read more about the component here:
+ * https://github.com/IvoLeist/dash_ngl
+ * Read more about the used WebGL protein viewer here:
+ * https://github.com/arose/ngl
+ */
+export default class DashNgl extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {stage: null, structuresList: []};
+    }
+
+    componentDidMount() {
+        const {id, stageParameters} = this.props;
+        const params = {...stageParameters};
+
+        const stage = new Stage(id, params);
```

Comment:
```suggestion
        const {id, stageParameters, viewportStyle} = this.props;
        const params = {...stageParameters};

        const stage = new Stage(id, params);
        stage.setSize(viewportStyle.width, viewportStyle.height);
```

Reply:
Thanks a lot ! for having a look into I integrated it and it works flawlessly :)